### PR TITLE
Add focus and hover colors to all themes in Item using functional variables

### PR DESCRIPTION
**Pull request by pksjce**
Before this PR, hardcoded RGB colors were being used for the hover and focus states. I changed them to use the functional color system as described in #1480 

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by pksjce** [comment]
Notes - 
1. hover -> `neutral.subtle`
2. focus -> `neutral.muted`
Would do well with some guidance from @auareyou here

**Comment by colebemis** [comment]
Looks like `neutral/danger.subtle` for the hover state and `neutral/danger.muted` for the focus state work in most cases. The exception is the focus state of the danger variant in dark high contrast:

CleanShot 2021-09-30 at 08 38 05

`danger.muted` and `danger.emphasis` are the same in dark high contrast which makes the `danger.fg` text invisible. 

Let's loop in @tallys and @auareyou because I think this issue is going to require more design work than I initially thought.

**Comment by pksjce** [comment]
Hey @auareyou @colebemis  - This still seems like an issue with the new colors in `dark_high_contrast` theme
Screen Shot 2021-11-15 at 3 21 56 pm

File: .changeset/serious-donkeys-clean.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@primer/components": patch
+---
+
+Add focus and hover colors to all themes in Item using functional variables
```
Comment (siddharthkp) [inline .changeset/serious-donkeys-clean.md:5]:
```suggestion
ActionList: Add focus and hover colors to all themes in Item using functional variables
```

File: src/ActionList/Item.tsx [hunk 2]
```diff
@@ -418,8 +391,6 @@ export const Item = React.forwardRef((itemProps, ref) => {
       data-id={id}
       onKeyPress={keyPressHandler}
       onClick={clickHandler}
-      hoverBackground={disabled ? 'inherit' : hoverBackground}
```
Comment (siddharthkp) [inline src/ActionList/Item.tsx:421]: this feels so good!

**Review by siddharthkp — APPROVED** [review]
(added a word for the automated changelog)

Looks good, let's ship it!

**Review by colebemis — APPROVED** [review]
🚢 🎉

**Pull request merged**